### PR TITLE
Lots of a11y fixes for NavList, round 2

### PR DESCRIPTION
**Pull request by camertron**
### Description

Fixes https://github.com/github/primer/issues/1120

This PR contains the following a11y fixes for `ActionList` and `NavList`:

1. Moves the "show more" item outside its group.
2. Converts the "show more" item into a button (currently a `<div>`).
3. Improves the text of "show more" items in previews to be more descriptive.
4. Removes an arg from the docs that's no longer used.
5. Attaches `aria-label`s to the `<nav>` element in all previews.
6. Raises an error if an `aria-label` is not provided.
7. Groups are now required to have headings.
8. Groups are labelled by headings and described by subtitles.
9. Added support for top-level items. Markup now dynamically changes depending on whether top-level items are present or the list contains only groups.
10. Added support for a top-level heading slot that will render an `<h2>` by default. Group headings render as `<h3>`s by default.

### Integration

> Does this change require any updates to code in production?

1. Any usages of `NavList`s that currently do not pass an `aria-label` will need to do so. 
2. The usages of `NavList` on the repo, org, and user settings pages that have 1-item groups should use top-level items instead.
3. Any `NavList` that is preceded by a heading should incorporate that heading into the component using the new top-level `heading` slot.

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Added/updated previews

### CODE REVIEW

File: .playwright/screenshots/snapshots.test.ts-snapshots/primer/alpha/nav_list/default.png
Comment (jonrohan) [inline .playwright/screenshots/snapshots.test.ts-snapshots/primer/alpha/nav_list/default.png:1]: Is this heading visual change going to effect the uses of NavList on dotcom?

Comment (camertron) [reply .playwright/screenshots/snapshots.test.ts-snapshots/primer/alpha/nav_list/default.png:1]: Yes and no. Dotcom already uses big `<h2>` headings above NavLists on settings pages, this change just brings them into the component. Instances in dotcom will still work without any changes, but we will want to change them anyway to add these headings and to remove 1-item groups like you see in the snapshot.

File: app/components/primer/alpha/nav_list.rb [hunk 1]
```diff
@@ -22,19 +22,48 @@ def self.custom_element_name
         "nav-list"
       end
 
+      renders_one :heading, lambda { |title:, heading_level: 2, **system_arguments|
```
[the remark this answers is not in the document]
Comment (camertron) [reply app/components/primer/alpha/nav_list.rb:25]: Yeah, this needs to be at the top level. Not sure those class names are appropriate in that case?

[the remark this answers is not in the document]
Comment (langermank) [reply app/components/primer/alpha/nav_list.rb:25]: I guess it doesn't need the other AL heading styles, still think its kind of confusing to offer a page level heading inside this component but lets see how it goes!

File: previews/primer/alpha/nav_list_preview.rb [hunk 2]
```diff
@@ -6,13 +6,11 @@ module Alpha
     class NavListPreview < ViewComponent::Preview
       # @label Playground
       def playground
-        render(Primer::Alpha::NavList.new(selected_item_id: :code_review_limits)) do |list|
-          list.with_group do |group|
-            group.with_heading(title: "Repository settings")
+        render(Primer::Alpha::NavList.new(aria: { label: "Repository settings" }, selected_item_id: :code_review_limits)) do |list|
+          list.with_heading(title: "Repository settings navigation")
```
Comment (langermank) [inline previews/primer/alpha/nav_list_preview.rb:10]:
```suggestion
          list.with_heading(title: "Repository settings")
```

I think this copy was suggested, but it doesn't fit well as a large h2 tag to say "navigation" and I think the design and use of `nav` makes it clear.

File: previews/primer/alpha/nav_list_preview.rb [hunk 3]
```diff
@@ -35,13 +33,11 @@ def playground
 
       # @label Default
       def default
-        render(Primer::Alpha::NavList.new(selected_item_id: :code_review_limits)) do |list|
-          list.with_group do |group|
-            group.with_heading(title: "Repository settings")
+        render(Primer::Alpha::NavList.new(aria: { label: "Repository settings" }, selected_item_id: :code_review_limits)) do |list|
+          list.with_heading(title: "Repository settings navigation")
```
Comment (langermank) [inline previews/primer/alpha/nav_list_preview.rb:37]:
```suggestion
          list.with_heading(title: "Repository settings")
```

File: previews/primer/alpha/nav_list_preview.rb [hunk 4]
```diff
@@ -62,21 +58,40 @@ def default
         end
       end
 
+      # @label Top-level items
+      #
+      def top_level_items
+        render(Primer::Alpha::NavList.new(aria: { label: "Account settings" })) do |list|
+          list.with_heading(title: "Account settings navigation")
```
Comment (langermank) [inline previews/primer/alpha/nav_list_preview.rb:65]:
```suggestion
          list.with_heading(title: "Account settings")
```

File: app/components/primer/alpha/nav_list.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (camertron) [reply app/components/primer/alpha/nav_list.rb:25]: Yeah I agree 😓 but based on the a11y feedback that's how it's supposed to be?

**Comment by camertron** [comment]
Dotcom integration branch (internal only): https://github.com/github/github/pull/271513

**Review by langermank — APPROVED** [review]

**Pull request merged**